From 7677ba7f1403c9a2f8c4a2843aa9bb95a57871b2 Mon Sep 17 00:00:00 2001
From: ghost <andreas.monzner@dream-property.net>
Date: Sat, 14 May 2016 13:00:41 +0200
Subject: [PATCH] TimerSanityCheck: fixed false positive "Channel not in
 service list" conflicts

(cherry picked from commit 922de81132b00b808acc2a080adbefdcb753bade)
---
 usr/lib/enigma2/python/Components/TimerSanityCheck.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/usr/lib/enigma2/python/Components/TimerSanityCheck.py b/usr/lib/enigma2/python/Components/TimerSanityCheck.py
index c214783..256afc9 100644
--- a/usr/lib/enigma2/python/Components/TimerSanityCheck.py
+++ b/usr/lib/enigma2/python/Components/TimerSanityCheck.py
@@ -184,12 +184,12 @@ class TimerSanityCheck:
 				if not fakeRecResult: # tune okay
 					feinfo = fakeRecService.frontendInfo().getFrontendData()
 					tunerType.append(feinfo.get("tuner_type"))
-				else: # tune failed.. so we must go another way to get service type (DVB-S, DVB-T, DVB-C)
+				else: # tune failed.. so we must go another way to get service type (iDVBFrontend.feSatellite, iDVBFrontend.feTerrestrial, iDVBFrontend.feCable)
 
 					def getServiceType(ref): # helper function to get a service type of a service reference
 						serviceInfo = serviceHandler.info(ref)
 						serviceInfo = serviceInfo and serviceInfo.getInfoObject(ref, iServiceInformation.sTransponderData)
-						return serviceInfo and serviceInfo["tuner_type"] or ""
+						return -1 if serviceInfo is None else serviceInfo.get("tuner_type", -1)
 
 					ref = timer.service_ref.ref
 					if ref.flags & eServiceReference.isGroup: # service group ?
-- 
1.9.1

